src/md.rs

```diff
@@ -1,19 +1,249 @@
+    is_separator, is_suspiciously_successive_range, is_thai, remove_accent, unicode_range,
 };
+use bitflags::{bitflags, Flags};
 use cached::proc_macro::cached;
+use cached::UnboundCache;
 use log::trace;
 use ordered_float::OrderedFloat;
+use unic::char::property::EnumeratedCharProperty;
+use unic::ucd::{is_white_space, GeneralCategory, Name};
 
 //
 // Mess detection module
 //
 
+// Mess Plugin Char representation
+// used to collect additional information about char
+// and eliminate repeated calculations
+
+#[derive(Copy, Clone, PartialEq)]
+pub struct MessDetectorCharFlags(u32);
+
+bitflags! {
+    impl MessDetectorCharFlags: u32 {
+        const WHITESPACE        = 0b0000_0000_0000_0000_0000_0000_0000_0001;
+        const UNPRINTABLE       = 0b0000_0000_0000_0000_0000_0000_0000_0010;
+        const SYMBOL            = 0b0000_0000_0000_0000_0000_0000_0000_0100;
+        const EMOTICON          = 0b0000_0000_0000_0000_0000_0000_0000_1000;
+        const COMMON_SAFE       = 0b0000_0000_0000_0000_0000_0000_0001_0000;
+        const WEIRD_SAFE        = 0b0000_0000_0000_0000_0000_0000_0010_0000;
+        const PUNCTUATION       = 0b0000_0000_0000_0000_0000_0000_0100_0000;
+        const SEPARATOR         = 0b0000_0000_0000_0000_0000_0000_1000_0000;
+        const ASCII             = 0b0000_0000_0000_0000_0000_0001_0000_0000;
+        const ASCII_ALPHABETIC  = 0b0000_0000_0000_0000_0000_0010_0000_0000;
+        const ASCII_GRAPHIC     = 0b0000_0000_0000_0000_0000_0100_0000_0000;
+        const ASCII_DIGIT       = 0b0000_0000_0000_0000_0000_1000_0000_0000;
+        const LATIN             = 0b0000_0000_0000_0000_0001_0000_0000_0000;
+        const ALPHABETIC        = 0b0000_0000_0000_0000_0010_0000_0000_0000;
+        const ACCENTUATED       = 0b0000_0000_0000_0000_0100_0000_0000_0000;
+        const CJK               = 0b0000_0000_0000_0000_1000_0000_0000_0000;
+        const HANGUL            = 0b0000_0000_0000_0001_0000_0000_0000_0000;
+        const KATAKANA          = 0b0000_0000_0000_0010_0000_0000_0000_0000;
+        const HIRAGANA          = 0b0000_0000_0000_0100_0000_0000_0000_0000;
+        const THAI              = 0b0000_0000_0000_1000_0000_0000_0000_0000;
+        const CASE_VARIABLE     = 0b0000_0000_0001_0000_0000_0000_0000_0000;
+        const LOWERCASE         = 0b0000_0000_0010_0000_0000_0000_0000_0000;
+        const UPPERCASE         = 0b0000_0000_0100_0000_0000_0000_0000_0000;
+        const NUMERIC           = 0b0000_0000_1000_0000_0000_0000_0000_0000;
+    }
+}
+
+#[derive(Copy, Clone)]
+pub struct MessDetectorChar {
+    pub character: char,
+    pub flags: MessDetectorCharFlags,
+    pub unicode_range: Option<&'static str>,
+}
+
+impl PartialEq for MessDetectorChar {
+    fn eq(&self, other: &Self) -> bool {
+        self.character == other.character
+    }
+}
+
+impl MessDetectorChar {
+    pub fn new(character: char) -> Self {
+        new_mess_detector_character(character)
+    }
+    fn in_category(
```

Comment:
This and many other fns are similar to those found in `util.rs` I wonder if one implementation could be replaced or extended for both.

Reply:
Functions from `uitls.rs` will be removed today.